From 9706d67f040a2ff1649397c25ad70d990925c119 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:50:01 -0400
Subject: [fs] nfsd: remove redundant check from nfsd4_check_resp_size

Message-id: <1403560314-32525-60-git-send-email-bfields@redhat.com>
Patchwork-id: 84480
O-Subject: [Patch RHEL7 059/172] nfsd4: remove redundant check from nfsd4_check_resp_size
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

cstate->slot and ->session are each set together in nfsd4_sequence.  If
one is non-NULL, so is the other.

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 1bed92cb3c7663240992f4d97cbe4d21783113a0
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4xdr.c b/fs/nfsd/nfs4xdr.c
index 2708fae..94b7f19 100644
--- a/fs/nfsd/nfs4xdr.c
+++ b/fs/nfsd/nfs4xdr.c
@@ -3583,8 +3583,6 @@ __be32 nfsd4_check_resp_size(struct nfsd4_compoundres *resp, u32 pad)
   return 0;
 
  session = resp->cstate.session;
- if (session == NULL)
-  return 0;
 
  if (xb->page_len == 0) {
   length = (char *)resp->p - (char *)xb->head[0].iov_base + pad;
-- 
1.7.1